### PR TITLE
Weather UI For Event Screen

### PR DESCRIPTION
**Pull request by arnaudlvq**
#### Overview
This PR introduces the UI implementation for Gregory's work on integrating weather API calls. The app handles three scenarios:
1. The event is a past event: no weather forecast is available.
2. The event is in more than 10 days: no weather forecast is available at the moment.
3. Loading state: displays the event in a card, one per day, sorted by date, with the forecast icon, wind speed, and temperature.

The implementation uses `HorizontalPager` to achieve a carousel-like UX design, providing an intuitive way for users to navigate through weather forecasts, displaying one day at a time. The paging is inspired from this [HorizontalPager documentation](https://developer.android.com/develop/ui/compose/layouts/pager?hl=fr#add-page).

### CODE REVIEW

**Comment by arnaudlvq** [comment]
https://github.com/Chimpagne/ChimpagneApp/assets/24866530/9109be18-05fc-4bac-8652-3144a3cd6657

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt:168]: a bit hacky but acceptable, since it's the same in french and english 😉

File: app/src/main/res/values/strings.xml
Comment (Athanaze) [inline app/src/main/res/values/strings.xml:169]: Fun touch :)

**Review by Athanaze — APPROVED** [review]

**Review by Kuriv2001 — CHANGES_REQUESTED** [review]
Well done! This will help users whether they should pack an umbrella or not and how to dress accordingly.

I tested all the possible cases you mentioned (past event, current event and far in the future event) with a location specified and they all seem to work perfectly.

I found a little edge case that you didn't consider. When there is no location set, then the weather says: "Failed to fetch weather data". You should either return an error message, that no location has been selected or make it mandatory to select a location for the event. This should be fixed in one of those ways.

After that, the PR can be approved.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt:87]: not translated

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt:183]: Is it really useful to create a component inside another one ? It is used only one time

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherCarousel.kt:116]: It is unclear if an external screen should use WeatherApp of WeatherCarousel

File: app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/ui/event/WeatherCarouselTest.kt
Comment (QuicheLorraine3600) [inline app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/ui/event/WeatherCarouselTest.kt:46]: will this test pass in the future ?

**Comment by Kuriv2001** [comment]
I agree with the comment of @QuicheLorraine3600 and created an issue for this task: #243. The PR is therefore approved.

**Review by Kuriv2001 — APPROVED** [review]

**Comment by QuicheLorraine3600** [comment]
Do not merge this PR, it breaks the tests

File: app/src/main/java/com/monkeyteam/chimpagne/ChimpagneApplication.kt
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/ChimpagneApplication.kt:7]: not part of this PR

File: app/src/main/java/com/monkeyteam/chimpagne/MainActivity.kt
Comment (QuicheLorraine3600) [inline app/src/main/java/com/monkeyteam/chimpagne/MainActivity.kt:81]: not part of this PR

**Review by QuicheLorraine3600 — CHANGES_REQUESTED** [review]
please remove offline mode related things for this PR

**Pull request closed without merging**
